src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddleware.cs

```diff
@@ -127,13 +129,34 @@ public async Task Invoke(HttpContext context)
         // Assumes the response headers have not been sent.  If they have, still attempt to write to the body.
         private Task DisplayException(ErrorContext errorContext)
         {
-            var compilationException = errorContext.Exception as ICompilationException;
-            if (compilationException != null)
+            var httpContext = errorContext.HttpContext;
+            var headers = httpContext.Request.GetTypedHeaders();
+            var acceptHeader = headers.Accept;
+
+            // If the client does not ask for HTML just format the exception as plain text
+            if (acceptHeader != null && !acceptHeader.Any(h => h.IsSubsetOf(new MediaTypeHeaderValue("text/html"))))
```

Comment:
Thank you 😁 

Doesn't this create a new MediaTypeHeaderValue inside a loop?

Reply:
Yea, the performance of this is probably really bad, though, we're on an exceptional path 😄

Reply:
I just fixed this performance bug!